lib/MailerInterface.php

```diff
@@ -2,34 +2,14 @@
 
 namespace Stampie;
 
-use Http\Client\HttpClient;
-
-/**
- * Takes a MailerInterface and sends to an AdapterInterface.
- *
- * @author Henrik Bjornskov <henrik@bjrnskov.dk>
- */
 interface MailerInterface
 {
     /**
-     * @param HttpClient $adapter
-     */
-    public function setHttpClient(HttpClient $adapter);
-
-    /**
-     * @param string $token
-     */
-    public function setServerToken($token);
-
-    /**
-     * @return string
-     */
-    public function getServerToken();
-
-    /**
+     * Sends an email message.
+     *
      * @param MessageInterface $message
      *
-     * @return bool
+     * @throws \Exception if an error happens during sending the message.
```

Comment:
Should we use `\Exception` here or introduce our exception interface? I'm asking this because Httplug has his own exception and this interface looks like `Http\Client\HttpClient` to me.

Reply:
We always throw our own exceptions (ApiException and HttpException) for clean failures of the API (and these are cases that you may want to catch)

Reply:
We may want to add an interface implemented by both classes though, for easier catching